From 8876e405055da08a60d5e8bb87f7b0e51c2d51b1 Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Sat, 16 Mar 2019 18:32:55 -0400
Subject: [PATCH] runqemu: suppress errors from "tput smam" when exiting

tmux and screen (TERM=screen-256color) don't seem to support tmut
smam/rmam options, and the stacktrace when exiting isn't nice.

Change subprocess.check_call to just call so that an exception isn't
raised if tput exits non-zero.
---
 scripts/runqemu | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/runqemu b/scripts/runqemu
index 1c96b29a40..76af7ed35e 100755
--- a/scripts/runqemu
+++ b/scripts/runqemu
@@ -1333,7 +1333,7 @@ def main():
             logger.info("SIGTERM received")
             os.kill(config.qemupid, signal.SIGTERM)
             config.cleanup()
-            subprocess.check_call(["tput", "smam"])
+            subprocess.call(["tput", "smam"])
         signal.signal(signal.SIGTERM, sigterm_handler)
 
         config.check_args()
@@ -1355,7 +1355,7 @@ def main():
         return 1
     finally:
         config.cleanup()
-        subprocess.check_call(["tput", "smam"])
+        subprocess.call(["tput", "smam"])
 
 if __name__ == "__main__":
     sys.exit(main())
-- 
2.21.0

